pkg/pillar/cmd/zedrouter/flowstats.go

```diff
@@ -670,8 +742,106 @@ func DNSStopMonitor(bnNum int) {
 	}
 }
 
+func checkDADProbe(ctx *zedrouterContext, bnNum int, packet gopacket.Packet) {
+	var needUpdate, foundSrcMac bool
+	var vifInfo []types.VifNameMac
+	var netstatus types.NetworkInstanceStatus
+	var vifTrig types.VifIPTrig
+
+	// use the IPAssigments of the NetworkInstanceStatus, since this is switched net
+	// and the field will not be assigned or modified by others
+	pub := ctx.pubNetworkInstanceStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		netstatus = st.(types.NetworkInstanceStatus)
+		if netstatus.Type != types.NetworkInstanceTypeSwitch || netstatus.BridgeNum != bnNum {
+			continue
+		}
+		vifInfo = netstatus.Vifs
+		break
+	}
+	if len(vifInfo) == 0 { // there is no Mac on the bridge
+		log.Tracef("checkDADProbe: no mac on the bridge")
+		return
+	}
+
+	var etherPkt *layers.Ethernet
+	etherLayer := packet.Layer(layers.LayerTypeEthernet)
+	if etherLayer != nil {
+		etherPkt, _ = etherLayer.(*layers.Ethernet)
+		for _, vif := range vifInfo {
+			if strings.Compare(etherPkt.SrcMAC.String(), vif.MacAddr) == 0 {
+				foundSrcMac = true
+				break
+			}
+		}
+	}
+	if !foundSrcMac {
+		log.Tracef("checkDADProbe: pkt no dst mac for us\n")
+		return
+	}
+
+	ip6Layer := packet.Layer(layers.LayerTypeIPv6)
+	if ip6Layer == nil {
+		return
+	}
+	ip6 := ip6Layer.(*layers.IPv6)
+	// We are looking for ICMPv6 Neighbor solicitation packet that
+	// tries to find if the address calculated locally is a duplicate.
+	// Such packets have a source IP of all zeroes (::)
+	if ip6.SrcIP.String() != "::" {
+		return
+	}
+
+	icmp6Layer := packet.Layer(layers.LayerTypeICMPv6NeighborSolicitation)
+	if icmp6Layer == nil {
+		return
+	}
+	for _, vif := range vifInfo {
+		if strings.Compare(vif.MacAddr, etherPkt.SrcMAC.String()) == 0 {
+			icmp6, _ := icmp6Layer.(*layers.ICMPv6NeighborSolicitation)
+			log.Tracef("ICMPv6: TargetAddress %s", icmp6.TargetAddress.String())
+
+			if _, ok := netstatus.IPAssignments[vif.MacAddr]; !ok {
```

Comment:
Here and in the else you are comparinhg against IPAssignments, but then you append to VifIPTrig.
Wouldn't it make sense to compare against VifIPTrig as well? Would make more sense logically.